lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1535,25 +1535,29 @@ def _write_soft_mask_groups(self):
     def hatchPattern(self, hatch_style):
         # The colors may come in as numpy arrays, which aren't hashable
         if hatch_style is not None:
-            edge, face, hatch = hatch_style
+            edge, face, hatch, lw = hatch_style
             if edge is not None:
                 edge = tuple(edge)
             if face is not None:
                 face = tuple(face)
-            hatch_style = (edge, face, hatch)
+            hatch_style = (edge, face, hatch, lw)
 
-        pattern = self.hatchPatterns.get(hatch_style, None)
+        pattern = self._hatch_patterns.get(hatch_style, None)
         if pattern is not None:
             return pattern
 
         name = next(self._hatch_pattern_seq)
-        self.hatchPatterns[hatch_style] = name
+        self._hatch_patterns[hatch_style] = name
         return name
 
+    hatchPatterns = _api.deprecated("3.8")(property(lambda self: {
```

Comment:
```suggestion
    hatchPatterns = _api.deprecated("3.10")(property(lambda self: {
```

An API change note is also required. See https://matplotlib.org/devdocs/devel/api_changes.html#deprecate-api

Reply:
Could you please check if the API change note is written correctly? Also the failing test is unrelated